### ISSUE TITLE
Can't override options of NativeKeyboard

### ISSUE THREAD

**Issue by HartHerbert** [EddyVerbruggen/cordova-plugin-native-keyboard#50]
I have an Ionic 3 application and I am trying to show the nativeKeyboard on a page (ChatPage).
At the first time I do this it works just fine but if i hide the messenger and setup the messenger again with different options, It doesn't override the older ones. I don't know if this is because I haven't paid it yet but It seems very strange to me and in the README it says that only the usage is restricted to 5 minutes, not that you can just show the messenger once.

```typescript
setupMessenger(){

    this.platform.ready().then(()=>{

      let options = {
        onSubmit: this.sendMessage,
        onKeyboardDidHide:this.onKeyboardHide,
        onKeyboardDidShow:null,
        onKeyboardWillHide:this.onKeyboardHide,
        onTextChanged:null,
        onKeyboardWillShow: this.onKeyboardShow,
        showKeyboard: false,
        type: "twitter",
        appearance: "light",
        secure: false,
        //autocorrectionEnabled: true,
        scrollToBottomAfterMessengerShows: true,
        autoscrollElement: this.content.getScrollElement(),
        keepOpenAfterSubmit: true,
        animated: true,
        text: "",
        placeholder: 'Schreib eine Nachricht',
        placeholderColor: "#DDDDDD",
        textViewBackgroundColor: "#FFFFFF",
        backgroundColor: "#F6F6F6",
        textViewBorderColor: "#F6F6F6",
        maxChars: 0,
        counterStyle: 'none',
        textColor: '#555555',

        leftButton: {
          type: 'ionicon',
          value: '\uf474',
          textStyle: 'normal',
          disabledWhenTextEntered: false,
          onPress: function () {

          }
        },
        rightButton: {
          type: 'ionicon',
          value: '\uf474',
          textStyle: 'normal',
          onPress: function () {

          }
        }
      };
      NativeKeyboard.showMessenger(options);
    });
  }
```

And this is my hide Method:

```typescript
hideNativeKeyboard(){
    let options = {
      animated: false
    };
    NativeKeyboard.hideMessenger(options);
  }
```

Both get called once:
```typescript
ionViewWillEnter(){
    this.platform.ready().then(()=>{
      this.setupMessenger();
    });
  }

ionViewWillLeave(){
    this.hideNativeKeyboard();
  }
```

And when I do it like this It works on the first chat page. for every other page it does not work.
Any idea what's going on ?
 Currently I am using the version `"version": "1.4.1"` of the nativeKeyboard

Environment information (if needed):
```
cli packages: (/Users/herberthart/Project/node_modules)

    @ionic/cli-plugin-cordova       : 1.6.1
    @ionic/cli-plugin-ionic-angular : 1.4.1
    @ionic/cli-utils                : 1.7.0
    ionic (Ionic CLI)               : 3.7.0

global packages:

    Cordova CLI : 7.0.1 

local packages:

    @ionic/app-scripts : 2.0.1
    Cordova Platforms  : ios 4.4.0
    Ionic Framework    : ionic-angular 3.5.3

System:

    Node       : v8.2.1
    OS         : macOS Sierra
    Xcode      : Xcode 8.3.3 Build version 8E3004b 
    ios-deploy : 1.9.1 
    npm        : 5.2.0 
```

**Comment by EddyVerbruggen** [comment]
Can I clone a repo to see this in action?

It's not related to having paid or not.

**Comment by HartHerbert** [comment]
I am working for a company and the sourcecode is not public, I'm sorry but there is nothing else I am doing with the Keyboard. But if you want I can post here the whole code of my page where I use the keyboard. I really just use it in this page.
Here is the .ts code of my ChatPage
```typescript
import {Component, NgZone, Renderer2, ViewChild} from '@angular/core';
import {Content, IonicPage, NavController, NavParams, Platform} from 'ionic-angular';
import {ChatMessage} from "../datatypes/chatMessage";
import {Keyboard} from "@ionic-native/keyboard";
import {ChatService} from "../../providers/chat-service";
import {GlobalState} from "../../providers/global-state";
import {DatabaseService} from "../../providers/database-service";

/**
 * Generated class for the ChatPage page.
 *
 * See http://ionicframework.com/docs/components/#navigation for more info
 * on Ionic pages and navigation.
 */

declare let NativeKeyboard:any;


@IonicPage()
@Component({
  selector: 'page-chat',
  templateUrl: 'chat.html',
})

export class ChatPage {

  @ViewChild(Content) content:Content;
  @ViewChild('chatmessages') list:any;

  messages:ChatMessage[];
  tthis = this;
  private ngZone:NgZone;
  public threadID:string;

  shouldScroll:boolean = true;

  constructor(public navCtrl: NavController, public navParams: NavParams,public platform:Platform,
              private renderer:Renderer2, private keyboard:Keyboard,
              public chatService:ChatService, public globalState:GlobalState, public databaseService:DatabaseService) {
    console.log("RAN OVER CONSTRUCTOR");
    this.messages = [];
    this.ngZone = new NgZone({enableLongStackTrace:false});
    this.threadID = "";//empty...

    //padding on the beginning

    if (this.platform.is('ios')) {
      this.keyboard.disableScroll(true);
    }
  }

  hideNativeKeyboard(){
    console.log("DESTROYING KEYBOARD");
    let options = {
      animated: false,
      onSubmit: null,
      onTextChanged: null,
      onKeyboardWillHide: null,
      onKeyboardDidHide: null,
      onKeyboardWillShow: null,
      onKeyboardDidShow: null,
      leftButton: null,
      rightButton: null,
      type: null,
      autoscrollElement: null,
      keepOpenAfterSubmit: null,
      scrollToBottomAfterMessengerShows: null,
      showKeyboard: null
    };
    NativeKeyboard.hideMessenger(options);
    //document.getElementsByTagName("body")[0].style.position = 'relative';
  }

  setScrollPadding(padding:number){
    this.renderer.setStyle(this.content.getScrollElement(),'padding-bottom',padding+'px');
    //this.content.getScrollElement().style.paddingBottom = padding+'px';
  }

  ngOnDestroy(){
    console.log("DESTROYING CHATPAGE", NativeKeyboard);
    this.hideNativeKeyboard();
  }


  sendMessage = (text:string) =>{
      //send message
    let new_message:ChatMessage = {};
    new_message.isOwn = true;
    new_message.author = this.globalState.username;
    new_message.text = text;
    new_message.time = new Date(Date.now());
    new_message.displayTime = new_message.time.getTime();
    new_message.id = Math.random().toString(36).substring(10);

    this.chatService.addMessageToThread(this.threadID,new_message);

    this.ngZone.run(()=>{
      this.messages.push(new_message);
      this.shouldScroll = true;
    });
    //TODO send new message to server
  }

  onKeyboardShow = (height:number) => {
    this.ngZone.run(()=>{
      //this.content.addScrollPadding(height + 5);
      this.setScrollPadding(height);
    });
    console.log("keyboard will show, height is: " + height);
  }

  onKeyboardHide = () => {
    this.ngZone.run(()=>{
      //this.content.addScrollPadding(10);
      this.setScrollPadding(10);
      this.shouldScroll = true;
      console.log("keyboard will hide");
    });
  }

  setupMessenger(){

    this.platform.ready().then(()=>{

      let options = {
        onSubmit: this.sendMessage,
        onKeyboardDidHide:this.onKeyboardHide,
        onKeyboardDidShow:null,
        onKeyboardWillHide:this.onKeyboardHide,
        onTextChanged:null,
        onKeyboardWillShow: this.onKeyboardShow,
        showKeyboard: false,
        type: "twitter",
        appearance: "light",
        secure: false,
        //autocorrectionEnabled: true,
        scrollToBottomAfterMessengerShows: true,
        autoscrollElement: this.content.getScrollElement(),
        keepOpenAfterSubmit: true,
        animated: true,
        text: "",
        placeholder: 'Schreib eine Nachricht',
        placeholderColor: "#DDDDDD",
        textViewBackgroundColor: "#FFFFFF",
        backgroundColor: "#F6F6F6",
        textViewBorderColor: "#F6F6F6",
        maxChars: 0,
        counterStyle: 'none',
        textColor: '#555555',

        leftButton: {
          type: 'ionicon',
          value: '\uf474',
          textStyle: 'normal',
          disabledWhenTextEntered: false,
          onPress: function () {

          }
        },
        rightButton: {
          type: 'ionicon',
          value: '\uf474',
          textStyle: 'normal',
          onPress: function () {

          }
        }
      };

      options.leftButton = null;
      NativeKeyboard.showMessenger(options);

      //document.getElementsByTagName("body")[0].style.position = 'fixed';

    });
  }


  ionViewWillEnter(){

    /*this.chatService.createNewThread().then((key:string)=>{
      console.log("Created new Thread KEY: ", key);
      this.threadID = key;
      this.chatService.addUserToDatabase();
      this.chatService.setDisconnect();
    }).catch((error:any)=>{
      //error while creating chat thread
      console.error("Error creating Thread: ", error);
    })*/

    this.threadID = "";
    this.threadID = this.navParams.get('thread').$key;

    this.platform.ready().then(()=>{
      this.setupMessenger();
    });

    console.log("Changed Thread ID to :", this.threadID);

    this.chatService.getMessagesFromThread(this.threadID).then((snapshots:any)=>{

      snapshots.forEach((snapshot:any)=>{
        let message:ChatMessage = {};
        message.id = snapshot.$key;
        message.author = snapshot.author;
        message.text = snapshot.text;
        if(message.author == this.globalState.username){
          message.isOwn = true;
        }else{
          message.isOwn = false;
        }
        message.displayTime = snapshot.time;
        message.time = new Date(snapshot.time);
        this.messages.push(message);
      })
    }).catch((error)=>{
      console.error("ERROR getting messages from thread:", error);
    })

    this.shouldScroll = true;

  }

  ionViewWillLeave(){
    this.hideNativeKeyboard();
  }

  ionViewDidLeave(){

  }

  scroll(){
    this.shouldScroll = false;
    setTimeout(()=>{
      this.content.scrollToBottom(0);
    },300);
  }

  ionViewDidEnter(){
    this.platform.ready().then(()=>{
      this.setScrollPadding(10);
      this.shouldScroll = true;
      this.chatService.checkIfOnline();
    });
  }

  ionViewDidLoad() {
    console.log('ionViewDidLoad ChatPage');
    this.shouldScroll = true;
  }
}
```

**Comment by HartHerbert** [comment]
I somehow found the error or whatever. 
The first time I call `NativeKeyboard.showMessenger(options);` everything works as expected.
But when I hide the Keyboard and then call `NativeKeyboard.showMessenger(options);` again the native function `- (void) configureMessenger { ... }` in `NKSLKViewController.m` does not get executed.

Could you take a look at it ?

**Comment by HartHerbert** [comment]
Just for you to know, I am using [WKWebview of the ionic-team](https://github.com/ionic-team/cordova-plugin-wkwebview-engine)

**Comment by EddyVerbruggen** [comment]
Thanks for the details and investigation!

**Comment by EddyVerbruggen** [comment]
Hi! THANKS for all the effort you put into reproducing this and helping me fix it, really top job 💯 

I think I've fixed the issue, so please update to 1.4.2 and see if that's correct.

**Comment by HartHerbert** [comment]
Working perfect now! Thanks Eddy, you're a hero !! 👍 💯
